src/tracks/playabletrack/wavetrack/ui/WaveClipTrimHandle.h

```diff
@@ -52,15 +63,26 @@ class WaveClipTrimHandle : public UIHandle
 
     HitTestPreview Preview(const TrackPanelMouseState& mouseState, AudacityProject* pProject) override;
 
-    virtual Result Click
+    Result Click
     (const TrackPanelMouseEvent& event, AudacityProject* pProject) override;
 
-    virtual Result Drag
+    Result Drag
     (const TrackPanelMouseEvent& event, AudacityProject* pProject) override;
 
-    virtual Result Release
+    Result Release
     (const TrackPanelMouseEvent& event, AudacityProject* pProject,
         wxWindow* pParent) override;
 
-    virtual Result Cancel(AudacityProject* pProject) override;
+    Result Cancel(AudacityProject* pProject) override;
+
+    // TrackPanelDrawable implementation
+
+    void Draw(TrackPanelDrawingContext &context,
+        const wxRect &rect,
+        unsigned iPass ) override;
+
+    wxRect DrawingArea(TrackPanelDrawingContext&,
+        const wxRect &rect,
+        const wxRect &panelRect,
+        unsigned iPass) override;
```

Comment:
See how SelectHandle overrides Enter, HasEscape, and Escape member functions.  These allow you to hit the Esc key and dismiss the yellow snap lines, enabling a fine, un-snapped adjustment, if that is what the user really wants.  The same might be done here.

Maybe some common base class, derived from UIHandle, should be defined?  Hm.

Reply:
You may note that I could have done the extra work I suggested in the UIHandle subclasses recently added to AdornedRulerPanel, but I didn't.

This is not mandatory extra work.  But I should reconsider how the pattern might be made easier and reusable across UIHandle classes that snap.

Reply:
> See how SelectHandle overrides Enter, HasEscape, and Escape member functions. These allow you to hit the Esc key and dismiss the yellow snap lines, enabling a fine, un-snapped adjustment, if that is what the user really wants. The same might be done here.
> 
> Maybe some common base class, derived from UIHandle, should be defined? Hm.

If I try to hit escape when using a select handle it just resets the current selection, with a debugger attached I see that both `::Escape` and `::Cancel` member functions are triggered, did I misunderstood something?

Reply:
You can hit escape to dismiss the yellow snap line before the click.  It seems you can't dismiss the yellow snap line while dragging.

Reply:
Does not seem to work as well, I still get snap lines during the selection